BuildMasterAutomation/Functions/Get-BMVariable.ps1

```diff
@@ -123,7 +125,10 @@ function Get-BMVariable
         [Object] $ServerRole,
 
         # Return the variable's value, not an object representing the variable.
-        [switch] $ValueOnly
+        [switch] $ValueOnly,
+
+        # Return the variable's value as an list. This flag must be used with the `ValueOnly` flag.
+        [switch] $AsList

```

Comment:
This isn't quite right because the variable won't always be an array/list, it might also be a map or scalar/string value.

We should have had this function always convert the BuildMaster variable to a native PowerShell type when returning it. Then it could have a `-Raw` switch that can be used to get the literal OtterScript variable expression. But implementing that now would be a breaking change.

I think a better way to do this (in a backwards-compatible manner) would be to add a `-AsPowerShellType` switch to the `Get-BMVariable` and `Set-BMVariable` functions so that we can simply pass PowerShell types back and forth (strings, arrays, and hashtables) to these functions and the module will handle converting it to/from OtterScript expressions when working with the BuildMaster API. We can create `ConvertTo-BMOtterScriptExpression` and `ConvertFrom-BMOtterScriptExpression` functions to handle this conversion.

@splatteredbits do you have any thoughts on this?

Reply:
I did make sure to take care of a case where the variable has to have the `@()` wrapper around it to ensure that it will return an array, but I do agree that having both the `ConvertTo-BMOtterScriptExpression` and `ConvertFrom-BMOtterScriptExpression` functions would be useful in general to aid in variable management.

Reply:
It's OK to make a breaking change. We'll just increment the major version to 4. I like the idea of converting the value from OtterScript types to native PowerShell types by default. Let's do that and increment the major version. I also like the `-Raw` switch to return the value as plaintext.